src/main/java/org/folio/service/dataimport/handlers/CreateOrderEventHandler.java

```diff
@@ -0,0 +1,325 @@
+  private Map<String, String> extractOkapiHeaders(DataImportEventPayload dataImportEventPayload) {
+    Map<String, String> headers = new HashMap<>();
+    headers.put(RestVerticle.OKAPI_HEADER_TENANT, dataImportEventPayload.getTenant());
+    headers.put(RestVerticle.OKAPI_HEADER_TOKEN, dataImportEventPayload.getToken());
+    headers.put(RestConstants.OKAPI_URL, dataImportEventPayload.getOkapiUrl());
+
+    String permissionsHeader = dataImportEventPayload.getContext().get(PERMISSIONS_KEY);
+    if (StringUtils.isNotBlank(permissionsHeader)) {
+      headers.put(OKAPI_PERMISSIONS_HEADER, permissionsHeader);
+    }
+    String userId = dataImportEventPayload.getContext().get(USER_ID_KEY);
+    if (StringUtils.isNotBlank(userId)) {
+      headers.put(RestVerticle.OKAPI_USERID_HEADER, userId);
+    }
+    return headers;
+  }
+
+  private Future<CompositePurchaseOrder> saveOrder(DataImportEventPayload dataImportEventPayload, String orderId,
+                                                   JsonObject tenantConfig, RequestContext requestContext) {
+    CompositePurchaseOrder orderToSave = Json.decodeValue(dataImportEventPayload.getContext().get(ORDER.value()), CompositePurchaseOrder.class);
+    orderToSave.setId(orderId);
+    orderToSave.setOrderType(CompositePurchaseOrder.OrderType.ONE_TIME); // todo: workaround for mapping profile
+    // in this handler a purchase order always is created in PENDING status despite the status that is set during mapping
+    orderToSave.setWorkflowStatus(WorkflowStatus.PENDING);
+
+    return purchaseOrderHelper.validateOrder(orderToSave, tenantConfig, requestContext)
+      .compose(errors -> {
+        if (CollectionUtils.isNotEmpty(errors)) {
+          return Future.failedFuture(new EventProcessingException(errors.toString()));
+        }
+        return purchaseOrderHelper.createPurchaseOrder(orderToSave, requestContext)
+          .onComplete(v -> dataImportEventPayload.getContext().put(ORDER.value(), Json.encode(orderToSave)))
+          .recover(e -> {
+            if (e instanceof HttpException) {
+              String message = ((HttpException) e).getError().getMessage();
+              if (message.contains(ID_UNIQUENESS_ERROR_MSG)) {
+                LOGGER.debug("saveOrder:: Failed to create order with existing id: '{}' due to duplicated event. Ignoring event processing", orderId);
+                return Future.failedFuture(new DuplicateEventException(message));
+              }
+            }
+            LOGGER.warn("saveOrder:: Error during creation order in the storage", e);
+            return Future.failedFuture(e);
+          });
+      });
+  }
+
+  private Future<CompositePoLine> saveOrderLines(String orderId, DataImportEventPayload dataImportEventPayload,
+                                                 JsonObject tenantConfig, RequestContext requestContext) {
+    CompositePoLine poLine = Json.decodeValue(dataImportEventPayload.getContext().get(ORDER_LINES_KEY), CompositePoLine.class);
```

Comment:
Now it is possible to save only one po line?

Reply:
during record import only one po line should be created per record. So far in the current PR a new order is created per every incoming record, but in subsequent changes the logic will be changed to create one order per multiple incoming records based on po lines limit per one order

Reply:
got it, thanks